### ISSUE TITLE
`EnforcedStyleForMultiline: comma` breaks `Style/TrailingCommaInArguments` after 1.75.4

### ISSUE THREAD

**Issue by Bo98** [rubocop/rubocop#14137]
## Expected behaviour

1.75.3 works correctly:

```console
$ bundle exec rubocop main.rb   
Inspecting 1 file
.

1 file inspected, no offenses detected
```

## Actual behaviour

A regression in 1.75.4 causes an error:

```console
$ bundle exec rubocop main.rb
For /private/tmp/rubocoptest: configuration from /private/tmp/rubocoptest/.rubocop.yml
Default configuration from /private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /private/tmp/rubocoptest/main.rb
An error occurred while Style/TrailingCommaInArguments cop was inspecting /private/tmp/rubocoptest/main.rb:3:0.
undefined method 'line' for nil
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cop/mixin/trailing_comma.rb:135:in 'RuboCop::Cop::TrailingComma#on_same_line?'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cop/mixin/trailing_comma.rb:131:in 'block in RuboCop::Cop::TrailingComma#no_elements_on_same_line?'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cop/mixin/trailing_comma.rb:131:in 'Array#each'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cop/mixin/trailing_comma.rb:131:in 'Enumerable#each_cons'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cop/mixin/trailing_comma.rb:131:in 'Enumerator#each'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cop/mixin/trailing_comma.rb:131:in 'Enumerable#none?'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cop/mixin/trailing_comma.rb:131:in 'RuboCop::Cop::TrailingComma#no_elements_on_same_line?'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cop/mixin/trailing_comma.rb:71:in 'RuboCop::Cop::TrailingComma#should_have_comma?'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cop/mixin/trailing_comma.rb:39:in 'RuboCop::Cop::TrailingComma#check_comma'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cop/mixin/trailing_comma.rb:25:in 'RuboCop::Cop::TrailingComma#check'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cop/style/trailing_comma_in_arguments.rb:107:in 'RuboCop::Cop::Style::TrailingCommaInArguments#on_send'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cop/commissioner.rb:107:in 'Kernel#public_send'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cop/commissioner.rb:107:in 'block (2 levels) in RuboCop::Cop::Commissioner#trigger_responding_cops'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cop/commissioner.rb:171:in 'RuboCop::Cop::Commissioner#with_cop_error_handling'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cop/commissioner.rb:106:in 'block in RuboCop::Cop::Commissioner#trigger_responding_cops'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cop/commissioner.rb:105:in 'Array#each'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cop/commissioner.rb:105:in 'RuboCop::Cop::Commissioner#trigger_responding_cops'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cop/commissioner.rb:69:in 'RuboCop::Cop::Commissioner#on_send'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-ast-1.44.1/lib/rubocop/ast/traversal.rb:20:in 'RuboCop::AST::Traversal#walk'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cop/commissioner.rb:87:in 'RuboCop::Cop::Commissioner#investigate'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cop/team.rb:174:in 'RuboCop::Cop::Team#investigate_partial'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cop/team.rb:108:in 'RuboCop::Cop::Team#investigate'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/runner.rb:348:in 'block in RuboCop::Runner#inspect_file'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/runner.rb:347:in 'Array#each'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/runner.rb:347:in 'Enumerable#flat_map'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/runner.rb:347:in 'RuboCop::Runner#inspect_file'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/runner.rb:290:in 'block in RuboCop::Runner#do_inspection_loop'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/runner.rb:324:in 'block in RuboCop::Runner#iterate_until_no_changes'
<internal:kernel>:168:in 'Kernel#loop'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/runner.rb:317:in 'RuboCop::Runner#iterate_until_no_changes'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/runner.rb:286:in 'RuboCop::Runner#do_inspection_loop'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/runner.rb:167:in 'block in RuboCop::Runner#file_offenses'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/runner.rb:192:in 'RuboCop::Runner#file_offense_cache'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/runner.rb:166:in 'RuboCop::Runner#file_offenses'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/runner.rb:154:in 'RuboCop::Runner#process_file'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/runner.rb:135:in 'block in RuboCop::Runner#each_inspected_file'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/runner.rb:134:in 'Array#each'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/runner.rb:134:in 'Enumerable#reduce'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/runner.rb:134:in 'RuboCop::Runner#each_inspected_file'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/runner.rb:120:in 'RuboCop::Runner#inspect_files'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/runner.rb:73:in 'RuboCop::Runner#run'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cli/command/execute_runner.rb:26:in 'block in RuboCop::CLI::Command::ExecuteRunner#execute_runner'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cli/command/execute_runner.rb:52:in 'RuboCop::CLI::Command::ExecuteRunner#with_redirect'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cli/command/execute_runner.rb:25:in 'RuboCop::CLI::Command::ExecuteRunner#execute_runner'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cli/command/execute_runner.rb:17:in 'RuboCop::CLI::Command::ExecuteRunner#run'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cli/command.rb:11:in 'RuboCop::CLI::Command.run'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cli/environment.rb:18:in 'RuboCop::CLI::Environment#run'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cli.rb:122:in 'RuboCop::CLI#run_command'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cli.rb:129:in 'RuboCop::CLI#execute_runners'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cli.rb:51:in 'block in RuboCop::CLI#run'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cli.rb:81:in 'RuboCop::CLI#profile_if_needed'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/lib/rubocop/cli.rb:43:in 'RuboCop::CLI#run'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/gems/rubocop-1.75.4/exe/rubocop:19:in '<top (required)>'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/bin/rubocop:25:in 'Kernel#load'
/private/tmp/rubocoptest/vgems/ruby/3.4.0/bin/rubocop:25:in '<top (required)>'
/opt/homebrew/Cellar/ruby/3.4.3/lib/ruby/3.4.0/bundler/cli/exec.rb:59:in 'Kernel.load'
/opt/homebrew/Cellar/ruby/3.4.3/lib/ruby/3.4.0/bundler/cli/exec.rb:59:in 'Bundler::CLI::Exec#kernel_load'
/opt/homebrew/Cellar/ruby/3.4.3/lib/ruby/3.4.0/bundler/cli/exec.rb:23:in 'Bundler::CLI::Exec#run'
/opt/homebrew/Cellar/ruby/3.4.3/lib/ruby/3.4.0/bundler/cli.rb:452:in 'Bundler::CLI#exec'
/opt/homebrew/Cellar/ruby/3.4.3/lib/ruby/3.4.0/bundler/vendor/thor/lib/thor/command.rb:28:in 'Bundler::Thor::Command#run'
/opt/homebrew/Cellar/ruby/3.4.3/lib/ruby/3.4.0/bundler/vendor/thor/lib/thor/invocation.rb:127:in 'Bundler::Thor::Invocation#invoke_command'
/opt/homebrew/Cellar/ruby/3.4.3/lib/ruby/3.4.0/bundler/vendor/thor/lib/thor.rb:538:in 'Bundler::Thor.dispatch'
/opt/homebrew/Cellar/ruby/3.4.3/lib/ruby/3.4.0/bundler/cli.rb:35:in 'Bundler::CLI.dispatch'
/opt/homebrew/Cellar/ruby/3.4.3/lib/ruby/3.4.0/bundler/vendor/thor/lib/thor/base.rb:584:in 'Bundler::Thor::Base::ClassMethods#start'
/opt/homebrew/Cellar/ruby/3.4.3/lib/ruby/3.4.0/bundler/cli.rb:29:in 'Bundler::CLI.start'
/opt/homebrew/Cellar/ruby/3.4.3/lib/ruby/gems/3.4.0/gems/bundler-2.6.8/exe/bundle:28:in 'block in <top (required)>'
/opt/homebrew/Cellar/ruby/3.4.3/lib/ruby/3.4.0/bundler/friendly_errors.rb:117:in 'Bundler.with_friendly_errors'
/opt/homebrew/Cellar/ruby/3.4.3/lib/ruby/gems/3.4.0/gems/bundler-2.6.8/exe/bundle:20:in '<top (required)>'
/Users/bo/.rbenv/versions/3.4.3/bin/bundle:25:in 'Kernel#load'
/Users/bo/.rbenv/versions/3.4.3/bin/bundle:25:in '<main>'
.

1 file inspected, no offenses detected

1 error occurred:
An error occurred while Style/TrailingCommaInArguments cop was inspecting /private/tmp/rubocoptest/main.rb:3:0.
configuration from /private/tmp/rubocoptest/.rubocop.yml
Errors are usually caused by RuboCop bugs.
Please, update to the latest RuboCop version if not already in use, and report a bug if the issue still occurs on this version.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.75.4 (using Parser 3.3.8.0, Prism 1.4.0, rubocop-ast 1.44.1, analyzing as Ruby 3.4, running on ruby 3.4.3) [arm64-darwin23]
Finished in 0.06399100017733872 seconds
```

## Steps to reproduce the problem

```yml
AllCops:
  NewCops: disable

Style/TrailingCommaInArguments:
  EnforcedStyleForMultiline: comma
```

```rb
# frozen_string_literal: true

Set[
  1,
  2,
]
```

## RuboCop version

```
$ [bundle exec] rubocop -V
1.75.4 (using Parser 3.3.8.0, Prism 1.4.0, rubocop-ast 1.44.1, analyzing as Ruby 3.4, running on ruby 3.4.3) [arm64-darwin23]
```

**Comment by bbatsov** [comment]
Seems it was casued by https://github.com/rubocop/rubocop/issues/14116. @viralpraxis Can you take a look at this one?

**Comment by viralpraxis** [comment]
Yeah, that's indeed caused by https://github.com/rubocop/rubocop/issues/14116. I'm working on a fix.

**Comment by viralpraxis** [comment]
Should be fixed by https://github.com/rubocop/rubocop/pull/14138
